### PR TITLE
add trino-python-client

### PR DESCRIPTION
**Pull request by h-vetinari**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Not sure if people here are aware, but presto recently [split](https://trino.io/blog/2020/12/27/announcing-trino.html) into presto (facebook-controlled) & trino (community-controlled). This adds the python client, intentionally using a name consistent with https://github.com/conda-forge/presto-python-client-feedstock.

### CODE REVIEW

**Comment by h-vetinari** [comment]
@conda-forge/staged-recipes
This should be good to go I think? 🤔

File: recipes/trino/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package:
+  name: trino-python-client
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/t/trino/trino-{{ version }}.tar.gz
+  sha256: 795d3b1a1bd0373ebf6af9a87ecc2aa6f4aa32e3098a8e05677d528a5324c098
+
+build:
+  # noarch: python
```
Comment (ocefpaf) [inline recipes/trino/meta.yaml:12]: Why not?

Comment (h-vetinari) [reply recipes/trino/meta.yaml:12]: Good point. I tend to want to try the different versions in CI explicitly before I switch on `noarch`

Comment (h-vetinari) [reply recipes/trino/meta.yaml]: Done, PTAL :)

**Pull request merged**